src/components/CheckBox.js

```diff
@@ -0,0 +1,68 @@
+import React, { useState } from 'react';
+import { View, Text, Image, StyleSheet } from 'react-native';
+import { RectButton } from 'react-native-gesture-handler';
+
+import { icons, theme } from '../constants';
+
+const { FONTS, COLORS } = theme;
+const BOX_SIZE = 11;
+
+const CheckBox = ({ label }) => {
+    const [ isChecked, setIsChecked ] = useState(false);
+
+    return (
+        <View style={styles.checkBoxContainer}>
+            <RectButton 
+                style={[styles.box, 
+                    { 
+                        backgroundColor: isChecked 
+                        ? COLORS.primary
+                        : COLORS.white,
+                        borderColor: isChecked
+                        ? COLORS.primary
+                        : COLORS.black
+                    }]}
+                onPress={() => setIsChecked(!isChecked)}
+            >
+                { isChecked ? 
+                    <Image 
+                        source={icons.check} 
+                        style={styles.icon}
+                        resizeMode="contain"
+                    />
+                    : <View/>
+                }
+            </RectButton>
+            <View style={styles.labelContainer}>
+                <Text style={styles.textLabel}>{ label }</Text>
+            </View>
+        </View>
+    )
+}
+
+export default CheckBox
+
+const styles = StyleSheet.create({
```

Comment:
You can create one or more files for the styles of your screens and components and import them where they are necessary, in this way the file will have only the components, the lines will be reduced and it will be easier to locate each thing.